cxf-soap/src/test/java/org/acme/cxf/soap/PojoClientTest.java

```diff
@@ -1,26 +1,33 @@
 package org.acme.cxf.soap;
 
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import javax.xml.namespace.QName;
+
 import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import jakarta.xml.ws.Service;
 import org.acme.cxf.soap.service.Address;
 import org.acme.cxf.soap.service.Contact;
 import org.acme.cxf.soap.service.ContactService;
 import org.acme.cxf.soap.service.ContactType;
 import org.acme.cxf.soap.service.NoSuchContactException;
-import org.apache.cxf.frontend.ClientProxyFactoryBean;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 @QuarkusTest
 public class PojoClientTest extends BaseTest {
 
     protected ContactService createCXFClient() {
-        String URL = getServerUrl() + "/cxf/services/contact";
-
-        ClientProxyFactoryBean factory = new ClientProxyFactoryBean();
-        factory.setServiceClass(ContactService.class);
-        factory.setAddress(URL);
-
-        return (ContactService) factory.create();
+        try {
+            final URL serviceUrl = new URL(getServerUrl() + "/cxf/services/contact?wsdl");
+            final QName qName = new QName(ContactService.TARGET_NS, ContactService.class.getSimpleName());
```

Comment:
Because otherwise the client seems to miss the info from the WSDL served via `/cxf/services/contact?wsdl` and those marshaling exceptions would occur. Not sure it does/should/would work also without the explicit namespace.

Reply:
In spring-boot it works (https://github.com/jboss-fuse/camel-spring-boot-examples/blob/camel-spring-boot-examples-3.20.1-branch/soap-cxf/src/test/java/sample/camel/PojoClientTest.java#L42) probably because https://github.com/jboss-fuse/camel-spring-boot-examples/blob/camel-spring-boot-examples-3.20.1-branch/soap-cxf/src/main/java/sample/camel/service/ContactService.java#L19 doesn't have the `@WebService` annotation ?

Reply:
@ffang Could you please check - if the web service has `@WebService` annotation and the `ContactService` created from `ClientProxyFactoryBean` (without QName specifying targetNS) should work ?

Reply:
Hi @llowinge ,

The difference comes from @WebService annotation indeed.
The class ClientProxyFactoryBean is used for Simple frontend(simple frontend means the SEI/Service Endpoint Interface is a POJO, no JAXWS annotation) to create a client proxy. As as ContactService now has JAXWS annotation, you should use JaxWsProxyFactoryBean which is from jaxws frontend to create the client proxy.

Freeman

Reply:
@ffang Thank you so much for clarification. It makes much more sense now to me. I've somehow missed the "Simple frontend" approach.